core/src/main/java/cn/edu/tsinghua/iotdb/benchmark/client/generate/SyntheticClient.java

```diff
@@ -57,115 +66,209 @@ public SyntheticClient(int id, CountDownLatch countDownLatch, CyclicBarrier barr
+                  break;
+                case RANGE_QUERY_ORDER_BY_TIME_DESC:
+                  status = dbWrapper.rangeQueryOrderByDesc((RangeQuery) query);
+                  break;
+                case VALUE_RANGE_QUERY_ORDER_BY_TIME_DESC:
+                  status = dbWrapper.valueRangeQueryOrderByDesc((ValueRangeQuery) query);
+                  break;
+                default:
+                  LOGGER.error("Unsupported operation type {}", operation);
+              }
+              statuses.add(status);
+            }
+            if (config.isIS_COMPARISON() && statuses.size() >= 2) {
+              Status status1 = statuses.get(0);
+              Status status2 = statuses.get(1);
+              boolean isError = false;
+              if (status1 != null
+                  && status2 != null
+                  && status1.getRecords() != null
+                  && status2.getRecords() != null) {
+                int point1 = status1.getQueryResultPointNum();
+                int point2 = status2.getQueryResultPointNum();
+                if (point1 != point2) {
+                  isError = true;
+                } else if (point1 != 0) {
+                  List<List<Object>> records1 = status1.getRecords();
+                  List<List<Object>> records2 = status2.getRecords();
+                  boolean needSort = true;
+                  if (query instanceof RangeQuery) {
+                    if (((RangeQuery) query).isDesc()) {
+                      needSort = false;
+                    }
                   }
-                }
-                if (needSort) {
-                  Collections.sort(records1, new RecordComparator());
-                  Collections.sort(records2, new RecordComparator());
-                }
-                // 顺序比较
-                for (int i = 0; i < point1; i++) {
-                  List<Object> record1 = records1.get(i);
-                  List<Object> record2 = records2.get(i);
-                  for (int j = 0; j < record1.size(); j++) {
-                    String r1 = String.valueOf(record1.get(j));
-                    String r2 = String.valueOf(record2.get(j));
-                    if (!r1.equals(r2)) {
-                      isError = true;
+                  if (needSort) {
+                    Collections.sort(records1, new RecordComparator());
+                    Collections.sort(records2, new RecordComparator());
+                  }
+                  // 顺序比较
+                  for (int i = 0; i < point1; i++) {
+                    List<Object> record1 = records1.get(i);
+                    List<Object> record2 = records2.get(i);
+                    for (int j = 0; j < record1.size(); j++) {
+                      String r1 = String.valueOf(record1.get(j));
+                      String r2 = String.valueOf(record2.get(j));
+                      if (!r1.equals(r2)) {
+                        isError = true;
```

Comment:
As soon as isError=true, we can break the loop, unless you need to count how many different points.

Reply:
OK, I think fast-fail is a good idea, I will fix it.